src/status_im/contexts/wallet/add_account/create_account/view.cljs

```diff
@@ -157,15 +159,29 @@
   [{:keys [on-change-text set-account-color set-emoji]
     {:keys [account-name account-color emoji]}
     :state}]
-  (let [on-auth-success (fn [password]
-                          (rf/dispatch
-                           [:wallet/import-and-create-keypair-with-account
-                            {:password            password
-                             :account-preferences {:account-name @account-name
-                                                   :color        @account-color
-                                                   :emoji        @emoji}}]))]
-    (fn [{:keys [customization-color keypair-name]}]
-      (let [{:keys [new-account-data]} (rf/sub [:wallet/create-account-new-keypair])]
+  (let [on-auth-success-mnemonic
+        (fn [password]
+          (rf/dispatch
+           [:wallet/import-and-create-keypair-with-account
+            {:password            password
+             :account-preferences {:account-name @account-name
+                                   :color        @account-color
+                                   :emoji        @emoji}}]))
+        on-auth-success-import-private-key
+        (fn [password]
+          (rf/dispatch
+           [:wallet/import-private-key-and-create-keypair-with-account
+            {:password            password
+             :account-preferences {:account-name @account-name
+                                   :color        @account-color
+                                   :emoji        @emoji}}]))]
```

Comment:
Better to extract them as separate functions, the new code style here looks hard to understand because of their long name, I know they depend on the atoms, but the atoms can be a parameter for these new functions :+1: